modules/util/portal-tools-service-builder/src/com/liferay/portal/tools/service/builder/dependencies/persistence_impl_finder_find.ftl

```diff
@@ -1179,6 +1157,76 @@ that may or may not be enforced with a unique index at the database level. Case
 					</#if>
 				</#list>
 
+				<#if finder.hasColumn("groupId") && finder.getColumn("groupId").hasArrayableOperator()>
+
+					<#list finderColsList as finderCol>
+						<#if finderCol.name == "groupId">
+
+							${finderCol.type}[] isNotEnabledGroupIds = new ${finderCol.type}[0];
+							${finderCol.type}[] isEnabledGroupIds = ${finderCol.names};
```

Comment:
Instead of removing values from `enabledGroupIds`, start both `enabledGroupIds` and `notEnabledGroupIds` as empty list as add values to the list as you loop through `groupIds`